src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionSetComparer.cs

```diff
@@ -1,77 +1,52 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Language.Intellisense;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
 {
     internal class SuggestedActionSetComparer : IComparer<SuggestedActionSet>
     {
-        private readonly SnapshotPoint? _targetPoint;
+        private readonly TextSpan? _targetSpan;
 
-        public SuggestedActionSetComparer(SnapshotPoint? targetPoint)
+        public SuggestedActionSetComparer(TextSpan? targetSpan)
         {
-            _targetPoint = targetPoint;
+            _targetSpan = targetSpan;
         }
 
-        private static int Distance(Span? textSpan, SnapshotPoint? targetPoint)
+        private static int Distance(Span? textSpan, TextSpan? targetSpan)
         {
             // If we don't have a text span or target point we cannot calculate the distance between them
-            if (!textSpan.HasValue || !targetPoint.HasValue)
+            if (!textSpan.HasValue || !targetSpan.HasValue)
             {
                 return int.MaxValue;
             }
 
             var span = textSpan.Value;
-            var position = targetPoint.Value.Position;
+            var target = targetSpan.Value;
 
-            if (position < span.Start)
-            {
-                return span.Start - position;
-            }
-            else if (position > span.End)
-            {
-                return position - span.End;
-            }
-            else
-            {
-                return 0;
-            }
+            // The distance of two spans is the sum of distance of their starts and their ends
```

Comment:
not sure i'm understanding that.  can you clarify it?  perhaps with some examples?

Reply:
`$ $ [  ]` has distance `2+3`
`$ [ ]$` has distance `1 + 0`
`$[ ]$` has distance `0 + 0`
`$ []  $` has distance `1 + 2`

Do you see the system? Essentially the simplest symetric span metric I could think of.